### ISSUE TITLE
Base class for optimizers

### ISSUE THREAD

**Issue by Oceania2018** [SciSharp/TensorFlow.NET#103]
This class defines the API to add Ops to train a model.  You never use this class directly, but instead instantiate one of its subclasses such as `GradientDescentOptimizer`, `AdagradOptimizer`, or `MomentumOptimizer`.
```cs
public abstract class Optimizer
{
}
```

**Comment by Esther2013** [comment]
`minimize` Add operations to minimize `loss` by updating `var_list`
`Optimizer.minimize` should return a `Operation`.
